File: app/src/main/java/com/project/scratchstudio/kith_andoid/GetInternalData/GetUserIdAndToken.java
package com.project.scratchstudio.kith_andoid.GetInternalData;

import android.content.SharedPreferences;
import android.util.Log;

import com.project.scratchstudio.kith_andoid.Activities.HomeActivity;

public class GetUserIdAndToken implements IGetInternalData {

    @Override
    public int get(SharedPreferences sp) {
        if( sp.getInt("cur_user_id", -1) == -1 ){
            return 1;
        }
        else {
            HomeActivity.createMainUser();
            HomeActivity.getMainUser().setId(sp.getInt("cur_user_id", -1));
            HomeActivity.getMainUser().setToken(sp.getString("cur_user_token", ""));
            return 2;
        }
    }

}

File: app/src/main/java/com/project/scratchstudio/kith_andoid/service/internal_storage/get_internal_data/GetUserIdAndToken.kt
package com.project.scratchstudio.kith_andoid.service.internal_storage.get_internal_data

import android.content.SharedPreferences
import com.project.scratchstudio.kith_andoid.activities.HomeActivity

class GetUserIdAndToken : IGetInternalData {

    override fun get(sp: SharedPreferences) {
        if (sp.getInt("cur_user_id", -1) != -1 && sp.getString("user_token", "") != "") {
            HomeActivity.createMainUser()
            HomeActivity.mainUser.id = sp.getInt("cur_user_id", -1)
            HomeActivity.mainUser.token = sp.getString("user_token", "") ?: ""
            HomeActivity.mainUser.password = sp.getString("cur_user_password", "") ?: ""
        }
    }

}
